system-tests/lib/cre/features/evm/v1/evm.go

```diff
@@ -0,0 +1,415 @@
+			if forwarderAddr == nil {
+				selectorsToDeploy = append(selectorsToDeploy, selector)
+			}
+		}
+
+		if len(selectorsToDeploy) > 0 {
+			deployErr := evm.DeployEVMForwarders(testLogger, creEnv.CldfEnvironment, selectorsToDeploy, creEnv.ContractVersions)
+			if deployErr != nil {
+				return nil, errors.Wrap(deployErr, "failed to deploy EVM Keystone forwarder")
+			}
+		}
+	}
+
+	tronForwardersSelectors, exist := chainsWithForwarders[blockchain.FamilyTron]
+	if exist {
+		deployErr := deployTronForwarders(testLogger, creEnv.CldfEnvironment, tronForwardersSelectors, creEnv.ContractVersions)
+		if deployErr != nil {
+			return nil, errors.Wrap(deployErr, "failed to deploy Tron Keystone forwarder")
+		}
+	}
+
+	// update node configs to include write-evm (evm v1) configuration
+	for _, donMetadata := range donsMetadata {
+		workerNodes, wErr := donMetadata.Workers()
+		if wErr != nil {
+			return nil, errors.Wrap(wErr, "failed to find worker nodes")
+		}
+
+		for _, workerNode := range workerNodes {
+			currentConfig := donMetadata.CapabilitiesAwareNodeSet().NodeSpecs[workerNode.Index].Node.TestConfigOverrides
+			updatedConfig, updErr := updateNodeConfig(workerNode, currentConfig, donMetadata.CapabilitiesAwareNodeSet().ChainCapabilities, creEnv)
+			if updErr != nil {
+				return nil, errors.Wrapf(updErr, "failed to update node config for node index %d", workerNode.Index)
+			}
+
+			donMetadata.CapabilitiesAwareNodeSet().NodeSpecs[workerNode.Index].Node.TestConfigOverrides = *updatedConfig
+		}
+	}
+
+	capabilities := make(map[uint64][]keystone_changeset.DONCapabilityWithConfig)
+	for _, donMetadata := range donsMetadata {
+		if capabilities[donMetadata.ID] == nil {
+			capabilities[donMetadata.ID] = []keystone_changeset.DONCapabilityWithConfig{}
+		}
+		for _, chainID := range donMetadata.CapabilitiesAwareNodeSet().ChainCapabilities[flag].EnabledChains {
+			fullName := corevm.GenerateWriteTargetName(chainID)
+			splitName := strings.Split(fullName, "@")
+
+			capabilities[donMetadata.ID] = append(capabilities[donMetadata.ID], keystone_changeset.DONCapabilityWithConfig{
+				Capability: kcr.CapabilitiesRegistryCapability{
+					LabelledName:   splitName[0],
+					Version:        splitName[1],
+					CapabilityType: 3, // TARGET
+					ResponseType:   1, // OBSERVATION_IDENTICAL
+				},
+				Config: &capabilitiespb.CapabilityConfig{},
+			})
+		}
+	}
+
+	return &cre.PreEnvStartupOutput{
+		DONCapabilityWithConfigs: capabilities,
+	}, nil
+}
+
+func (o *EVM) PostEnvStartup(
+	ctx context.Context,
+	testLogger zerolog.Logger,
+	creEnv *cre.Environment,
+) error {
+	dons := creEnv.DonTopology.DonsWithFlag(flag)
+	if len(dons) == 0 {
+		return nil
+	}
+
+	consensusVersion := "v1"
+	consensusDON, oneErr := creEnv.DonTopology.OneDonWithFlag(cre.ConsensusCapability)
```

Comment:
This should be:
- loop over every DON with consensus (v1 or v2)
- update the forwarder with 1 config per DON w consensus